ui/src/components/create_client/CreateMobileClientBaseClass.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from 'react';
+import { Grid, Form } from 'patternfly-react';
+import { CREATE_CLIENT_APP_ID, CREATE_CLIENT_NAME } from './Constants';
+import { VerticalFormField } from './VerticalFormField';
+
+/**
+ * Component for the Android specific create mobile client form.
+ */
+class CreateMobileClientBaseClass extends Component {
+  constructor(platformName, props) {
+    super(props);
+    this.state = {
+      valid: false,
+      validationState: {}
+    }
+    this.config = {
+      platform: platformName,
+      appName: {
+        label: '* App Name',
+        example: 'myapp',
+        help: 'App name must match ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
+      },
+      appIdentifier: {
+        label: '* Package Name',
+        example: 'org.aerogear.myapp',
+        help: 'Package name must match ^[a-zA-Z][\\w]*(\\.[a-zA-Z][\\w]*)+$',
```

Comment:
the same here.

Reply:
I agree, however here it is a bit more complicated than that.
The appIpdentifier

- Must have at least two segments (one or more dots).
- Each segment must start with a letter.
- All characters must be alphanumeric or an underscore [a-zA-Z0-9_].